packages/flutter_tools/lib/src/commands/build_web.dart

```diff
@@ -87,6 +94,10 @@ class BuildWebCommand extends BuildSubCommand {
     if (buildInfo.isDebug) {
       throwToolExit('debug builds cannot be built directly for the web. Try using "flutter run"');
     }
+    if(!(stringArg('base-href').startsWith('/') && stringArg('base-href').endsWith('/')))
```

Comment:
Why do we enforce this? 

Some use-cases might benefit from setting the base-href to "" or "./".

Reply:
My understanding is that setting `base-href` to "" or "./" is equivalent to omitting the `base-href` altogether. Are there cases where that's not true?

Reply:
Some devices based on web technologies (eg WebOS) don't require it. I think this should not be enforced. I would be happy with any of the following changes:
(1) Allow the tool to remove the tag
(2) Allow setting the base-href to "" or "./"
(3) Remove the error and at most include a warning

Reply:
> (1) Allow the tool to remove the tag

You can remove the `<base>` tag from your `web/index.html` and the tool won't add it back. Would that work?Removing the `<base>` tag will give the same result as `<base href="">` if I understand correctly.

Reply:
Thanks for the reply.

I guess that can work. Is there a way to remove this automatically by using the tool (ie not requiring additional developer scripts)?

Reply:
When you create the flutter project, you'll find a `web/index.html` file that's auto-generated for you. You own that file from that point on. So just remove the `<base href="$FLUTTER_BASE_HREF">` from there and you should be good to go.

Reply:
@mdebbar sorry for the late reply. That is a valid solution and works fine. One could do it manually or add a script to do so but the developer experience would be much better if the developer didn't require writing their own script and instead be able to use the built-in command. What do you think?

Reply:
Could you clarify what you need the script for? I'm not sure I understand.

If you remove it once from `web/index.html`, does that not work?